operations/helm/charts/mimir-distributed/templates/_helpers.tpl

```diff
@@ -429,6 +433,35 @@ Examples:
 {{- $section | toYaml -}}
 {{- end -}}
 
+{{/*
+Return the Vault Agent pod annotations if enabled and required by the component
+*/}}
+{{- define "mimir.vaultAgent.annotations" -}}
+{{- $vaultEnabledComponents := dict
+  "admin-api" true
+  "alertmanager" true
+  "compactor" true
+  "distributor" true
+  "gateway" true
+  "ingester" true
+  "overrides-exporter" true
+  "querier" true
+  "query-frontend" true
+  "query-scheduler" true
+  "ruler" true
+  "store-gateway" true
+-}}
+{{- if hasKey $vaultEnabledComponents . }}
+vault.hashicorp.com/agent-inject: 'true'
+vault.hashicorp.com/role: 'gem'
```

Comment:
Is the role really `gem`? Is this applicable only for GEM or also open source mimir? If only for GEM, we need to add more condition to make sure this annotation only shown when enterprise is enabled.